### PR TITLE
day 8 python 

### PR DESCRIPTION
**Pull request by RongkunWang**
Given the nature of the inputs, the problem is reduced down to finding Least common multiple for a bunch of prime, so I time them together.

### CODE REVIEW

File: src/python/08_rk.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+        # size of the loop first
+        l_good_endings.append([len(d_nstep_from_start) - d_nstep_from_start.index(current), set(l_step_ending_in_Z)])
+        pass
+
+    l_sets = [set(i[1]) for i in l_good_endings]
+    #  print("MAP", dmap)
+    #  print(l_good_endings)
+    #  print("START", len(seq))
+    # hacky answer due to the nature of the problem set:
+    return reduce(lambda x, y : x * y, [i[0] for i in l_good_endings]) * len(seq)
```
Comment (Moelf) [inline src/python/08_rk.py:81]: can this just be
```python
return reduce(*, [i[0] for i in l_good_endings]) * len(seq)
```

Comment (RongkunWang) [reply src/python/08_rk.py:81]: last line? I didn't think of it, it looks like you're in julia too long
```
  File "/Users/rowang/Library/CloudStorage/GoogleDrive-elviswang011@gmail.com/My Drive/AdventOfCode/AoC2023/src/python/08_rk.py", line 82
    return reduce(*, [i[0] for i in l_good_endings]) * len(seq)
                   ^
SyntaxError: invalid syntax
```

Comment (Moelf) [reply src/python/08_rk.py:81]: 🤡

Comment (RongkunWang) [reply src/python/08_rk.py:81]: * is am operator. looks like reduce takes a function

**Comment by Cryptum169** [comment]
Do you mind explaining your approach? I don't agree with the `LCM` way but I also don't quite understand what your approach here is, thanks!

**Comment by RongkunWang** [comment]
I was scratching my head to try to reduce the complexity of the problem as I was thinking maybe there is multiple XXZ on the loop.

I did some printout and saw when it takes N step for each start XXA to arrive at the only XXZ and it always loop back to the position righr after the start, this means the loop size is also N. So the solution is a1N1=a2N2=...amNm for all m starts. And an LCM could be used. 

But then I noticed all N are prime numbers, I was very lazy at the point and just times everything together to get the LCM.

Not the most traditional programming problem I'd say :😀

**Comment by Cryptum169** [comment]
Fair enough, thanks for the explanation tho :)

**Pull request merged**